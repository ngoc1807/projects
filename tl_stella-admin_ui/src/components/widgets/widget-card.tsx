import { FC, ReactElement } from "react";
import { Card, CardContent, useTheme } from "@mui/material";

interface Props {
  children: ReactElement;
}

export const WidgetCard: FC<Props> = ({ children }) => {
  const theme = useTheme();
  return (
    <Card
      sx={{
        borderTopRightRadius: 55,
        marginTop: 2,
        padding: 0,
        paddingRight: 0,
        paddingBottom: 0,
        position: "relative",
        minHeight: "100%",
        borderBottomWidth: 1,
        borderBottomColor: theme.palette.divider,
        borderBottomStyle: "solid",
        "&::before": {
          content: '""',
          position: "absolute",
          backgroundColor: "secondary.main",
          width: "110px",
          height: 2,
          top: 0,
          left: 0,
        },
      }}
      raised={false}
      elevation={0}
      square
    >
      <CardContent
        sx={{
          paddingRight: "32px",
          "&:last-child": {
            paddingBottom: "32px",
          },
        }}
      >
        {children}
      </CardContent>
    </Card>
  );
};
