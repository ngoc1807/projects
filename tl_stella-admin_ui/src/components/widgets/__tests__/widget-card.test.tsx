import { cleanup, render } from "@testing-library/react";
import { WidgetCard } from "../widget-card";

describe("widget-card", () => {
  afterEach(() => {
    cleanup();
  });
  it("should render correctly", () => {
    const { container } = render(
      <WidgetCard>
        <div>Foo Div</div>
      </WidgetCard>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
