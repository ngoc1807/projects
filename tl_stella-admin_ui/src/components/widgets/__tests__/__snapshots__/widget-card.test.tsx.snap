// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`widget-card should render correctly 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiCard-root css-1surxgo-MuiPaper-root-MuiCard-root"
>
  <div
    class="MuiCardContent-root css-1ioqiuq-MuiCardContent-root"
  >
    <div>
      Foo Div
    </div>
  </div>
</div>
`;
